polyfill/lib/intl.mjs

```diff
@@ -205,45 +205,113 @@ function hasTimeOptions(options) {
   return 'hour' in options || 'minute' in options || 'second' in options;
 }
 
-function extractOverrides(datetime, main) {
-  let formatter, calendar;
-  const Instant = GetIntrinsic('%Temporal.Instant%');
-  const Date = GetIntrinsic('%Temporal.Date%');
+function extractOverrides(temporalObj, main) {
   const DateTime = GetIntrinsic('%Temporal.DateTime%');
-  const MonthDay = GetIntrinsic('%Temporal.MonthDay%');
-  const Time = GetIntrinsic('%Temporal.Time%');
-  const YearMonth = GetIntrinsic('%Temporal.YearMonth%');
-
-  if (datetime instanceof Time) {
-    datetime = datetime.toDateTime(new Date(1970, 1, 1));
-    formatter = main[TIME];
-  }
-  if (datetime instanceof YearMonth) {
-    calendar = datetime.calendar.id;
-    const { isoYear, isoMonth, refISODay } = datetime.getISOFields();
-    datetime = new Date(isoYear, isoMonth, refISODay, datetime.calendar);
-    formatter = main[YM];
-  }
-  if (datetime instanceof MonthDay) {
-    calendar = datetime.calendar.id;
-    const { refISOYear, isoMonth, isoDay } = datetime.getISOFields();
-    datetime = new Date(refISOYear, isoMonth, isoDay, datetime.calendar);
-    formatter = main[MD];
-  }
-  if (datetime instanceof Date) {
-    calendar = calendar || datetime.calendar.id;
-    datetime = datetime.toDateTime(new Time(12, 0));
-    formatter = formatter || main[DATE];
-  }
-  if (datetime instanceof DateTime) {
-    calendar = calendar || datetime.calendar.id;
-    formatter = formatter || main[DATETIME];
-    datetime = main[TIMEZONE].getInstantFor(datetime);
-  }
-  if (datetime instanceof Instant) {
-    formatter = formatter || main[DATETIME];
-    return { absolute: datetime, formatter, calendar };
-  } else {
-    return {};
+
+  if (ES.IsTemporalTime(temporalObj)) {
+    const hour = GetSlot(temporalObj, HOUR);
+    const minute = GetSlot(temporalObj, MINUTE);
+    const second = GetSlot(temporalObj, SECOND);
+    const millisecond = GetSlot(temporalObj, MILLISECOND);
+    const microsecond = GetSlot(temporalObj, MICROSECOND);
+    const nanosecond = GetSlot(temporalObj, NANOSECOND);
+    const datetime = new DateTime(1970, 1, 1, hour, minute, second, millisecond, microsecond, nanosecond, main[CAL_ID]);
+    return {
+      instant: main[TIMEZONE].getInstantFor(datetime),
```

Comment:
Is `main[TIMEZONE]` equal to the timezone that Intl.DateTimeFormat uses to convert the timestamp back into a date/time tuple?

Reply:
As far as I can tell from reading the spec, it is. (It's equal to `Temporal.TimeZone.from(main.resolvedOptions().timeZone)`.)